### PR TITLE
Add StyleSheet.flatten

### PR DESCRIPTION
**Pull request by mroswald**
adding StyleSheet.flatten

### CODE REVIEW

**Comment by mroswald** [comment]
rebased to current master

**Comment by mroswald** [comment]
@lelandrichardson anything missing here?

**Comment by lelandrichardson** [comment]
@mroswald thanks for the contribution! this isn't really doing any flattening though...

`StyleSheet.flatten` needs to allow arbitrarily nested arrays of styles.

```
flatten([foo, [bar, baz], bux])
```

is roughly equivalent to:

```
Object.assign({}, foo, bar, baz, bux])
```

**Comment by mroswald** [comment]
@lelandrichardson Ok, I fixed that and added the test cases specified in the React Native documentation: https://facebook.github.io/react-native/docs/stylesheet.html

File: src/api/StyleSheet.js [hunk 1]
```diff
@@ -5,6 +5,13 @@ const StyleSheet = {
   create(styles) {
     return styles;
   },
+  flatten(styles) {
+    if (Array.isArray(styles)) {
+      return Object.assign({}, ...styles);
```
Comment (lelandrichardson) [inline src/api/StyleSheet.js]: quick tweak:

```
return Object.assign({}, ...styles.map(flatten))
```

Which should make this pass for nested style arrays.

**Comment by lelandrichardson** [comment]
@mroswald thanks for the quick turnaround! I had one quick suggestion if you don't mind.

**Comment by mroswald** [comment]
@lelandrichardson aye! :-)

**Pull request merged**